src/neuroconv/tools/testing/data_interface_mixins.py

```diff
@@ -128,29 +176,64 @@ def check_align_starting_time_internal(self):
 
-        aligned_timestamps = unaligned_timestamps + 1.23 + np.random.random(size=unaligned_timestamps.shape)
-        self.interface.align_timestamps(aligned_timestamps=aligned_timestamps)
+        aligned_timestamps = pre_alignment_original_timestamps + 1.23
+        fresh_interface.align_timestamps(aligned_timestamps=aligned_timestamps)
 
-        retrieved_aligned_timestamps = self.interface.get_timestamps()
-        assert_array_equal(x=retrieved_aligned_timestamps, y=aligned_timestamps)
+        post_alignment_original_timestamps = fresh_interface.get_original_timestamps()
+        assert_array_equal(x=post_alignment_original_timestamps, y=pre_alignment_original_timestamps)
 
-    def check_align_starting_time_external(self):
-        pass  # TODO: generalize
+    def check_interface_interpolation(self):
+        """Mimic the tests in the 'test_temporal_alignment_methods.py'."""
+        # This interface's system starts 3.23 seconds after the session start time
+        interface_system_delayed_start = 3.23
 
-    def check_temporal_alignment(self):
-        self.check_get_timestamps()
-        self.check_align_starting_time_internal()
-        self.check_align_starting_time_external()
+        # The clock on this interface's system is slightly slower than the main time stream
+        interface_system_average_delay_per_second = 0.0001
 
-    def run_custom_checks(self):
-        """Override this in child classes to inject additional custom checks."""
-        pass
+        # The drift from the trial tracking system adds up over time
+        interface_system_total_drift = interface_system_delayed_start * interface_system_average_delay_per_second
 
-    def test_conversion_as_lone_interface(self):
+        # Timing of secondary events stream according to this interface's system
+        unaligned_behavior_event_times = [5.6, 7.3, 8.7]
+        events_interface = MockBehaviorEventInterface(event_times=unaligned_behavior_event_times)
+
+        fresh_interface = self.data_interface_cls(**self.test_kwargs)
+        unaligned_timestamps = fresh_interface.get_original_timestamps()
+
+        aligned_timestamps = (  # use linspace to match the exact length of timestamps
+            unaligned_timestamps
+            + interface_system_delayed_start
+            + np.linspace(start=0.0, stop=interface_system_total_drift, num=len(unaligned_timestamps))
+        )
+        fresh_interface.align_timestamps(aligned_timestamps=aligned_timestamps)
+
+        events_interface.align_by_interpolation(
+            unaligned_timestamps=unaligned_timestamps, aligned_timestamps=aligned_timestamps
+        )
+
+        aligned_behavior_event_times = events_interface.get_timestamps()
+        expected_aligned_behavior_event_times = [8.830632, 10.530796, 11.930964]
+        assert_array_almost_equal(x=aligned_behavior_event_times, y=expected_aligned_behavior_event_times)
+
+    def check_interface_temporal_alignment(self):
```

Comment:
As you mentioned I do find this level indirection unecessary (better just to list all the things that you are going to test below) but I think you want to use this as organization for separating the properties that appended to the nwb from the ones that write to the nwb file.That seems fine but I still find the naming `internal` confusing with this idea. Maybe I am not getting it right.

Reply:
Where is `internal` used anymore? I thought I removed it

Reply:
Oh I see it in the docstring

Reply:
Fixed